packages/@aws-cdk/aws-apigateway/lib/method.ts

```diff
@@ -360,7 +360,12 @@ export class Method extends Resource {
     }
 
     if (options.requestValidatorOptions) {
-      const validator = (this.api as RestApi).addRequestValidator('validator', options.requestValidatorOptions);
+      let id = `${this.resource.path}${this.httpMethod}Validator`;
+      // For backward compatibility
+      if (options.requestValidatorOptions.requestValidatorName) {
```

Comment:
I'm not sure how this achieves backwards compatibility. We need to add a feature flag for this, since it's a breaking change.